makeflow/src/makeflow_catalog_reporter.c

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright (C) 2016- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "dag.h"
+#include "timestamp.h"
+#include "stringtools.h"
+#include "list.h"
+#include "makeflow_summary.h"
+#include "catalog_query.h"
+#include "json.h"
+#include "json_aux.h"
+#include "username.h"
+#include "batch_job.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+
+//packet for start and stop, once per min, configurable time
+
+//make sure to send: project name, owner, type
+
+//waiting means: inputs not ready, running means: took job and submitted
+
+int makeflow_catalog_summary(struct dag* d, char* name, batch_queue_type_t type){
+    struct dag_node *n;
+    const char *fn = 0;
+    dag_node_state_t state;
+    
+    struct list *failed_tasks; 
+    failed_tasks = list_create();
+    
+    int tasks_completed = 0;
+    int tasks_aborted   = 0;
+    int tasks_waiting   = 0;
+    int tasks_running   = 0;
+
+    for (n = d->nodes; n; n = n->next) {
+        state = n->state;
+        if (state == DAG_NODE_STATE_FAILED && !list_find(failed_tasks, (int (*)(void *, const void *)) string_equal, (void *) fn))
+            list_push_tail(failed_tasks, (void *) n->command);
```

Comment:
If I read this correctly, you are keeping a list of failed tasks, so as to de-duplicate them.
I think you can just do `tasks_failed++` here, since each makeflow node is already unique.
(It is possible for two different tasks to have the same command string, since the file namespace can vary per task.)